src/main/java/org/testng/internal/annotations/TestAnnotation.java

```diff
@@ -22,6 +22,7 @@
   private boolean m_singleThreaded = false;
   private Class<?> m_dataProviderClass = null;
   private IRetryAnalyzer m_retryAnalyzer = null;

```

Comment:
I will get rid of that attribute when the deprecated code path is removed. Its easy that way.

Reply:
But we don't need it anymore

Reply:
@juherr - We need it to ensure backward compatibility. Third party clients who are extending `org.testng.internal.TestNGMethod` will still be traversing that codepath because our interface has new additions but those methods are `default` in nature.

There's nothing wrong in leaving it there and removing it off when the deprecated code paths are going to be removed no ?